taca/illumina/__init__.py

```diff
@@ -0,0 +1,172 @@
+""" Submodule with Illumina-related code
+"""
+import os
+
+from datetime import datetime
+from xml.etree import ElementTree as ET
+
+from taca.log import LOG
+from taca.utils import misc
+from taca.utils.config import CONFIG
+from taca.utils.filesystem import chdir
+
+
+def demultiplex_HiSeq_X(run):
+    """ Demultiplexing for HiSeq X runs
+    """
+    LOG.info('Building bcl2fastq command')
+    config = CONFIG['preprocessing']
+    with chdir(run):
+        cl_options = config['bcl2fastq']
+        cl = [cl_options.get('XTen')]
+
+        # Main options
+        if cl_options.get('runfolder'):
+            cl.extend(['--runfolder', cl_options.get('runfolder')])
+        cl.extend(['--output-dir', cl_options.get('output-dir', 'Demultiplexing')])
+
+        # Advanced options
+        if cl_options.get('input-dir'):
+            cl.extend(['--input-dir', cl_options.get('input-dir')])
+        if cl_options.get('intensities-dir'):
+            cl.extend(['--intensities-dir', cl_options.get('intensities-dir')])
+        if cl_options.get('interop-dir'):
+            cl.extend(['--interop-dir', cl_options.get('interop-dir')])
+        if cl_options.get('stats-dir'):
+            cl.extend(['--stats-dir', cl_options.get('stats-dir')])
+        if cl_options.get('reports-dir'):
+            cl.extend(['--reports-dir', cl_options.get('reports-dir')])
+
+        # Processing cl_options
+        threads = cl_options.get('loading-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--loading-threads', '{}'.format(threads)])
+        threads = cl_options.get('demultiplexing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--demultiplexing-threads', '{}'.format(threads)])
+        threads = cl_options.get('processing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--processing-threads', '{}'.format(threads)])
+        threads = cl_options.get('writing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--writing-threads', '{}'.format(threads)])
+
+        # Behavioral options
+        adapter_stringency = cl_options.get('adapter-stringency')
+        if adapter_stringency and type(adapter_stringency) is float:
```

Comment:
hmmm this code is old, why didn't you told me before!! hehe

you're right, tack tack!!